chapters/modules/chapter_8/pages/sharp.adoc

```diff
@@ -0,0 +1,78 @@
+
+== Recursive Proofs in the SHARP
+
+One of the most exciting aspects of the SHARP is its use of recursive proofs. Instead of sending the generated proofs directly to the Solidity Verifier, the proofs are first sent to a STARK Verifier program written in Cairo. This STARK Verifier then generates a new proof that the initial proofs were verified. These can be sent back into the SHARP and the STARK Verifier.
+
+This process continues recursively, with each new proof being sent to the Cairo Verifier until a trigger is reached. At this point, the last proof in the series is sent to the Solidity Verifier on Ethereum. This approach allows for greater parallelization of the computation and reduces the time and cost associated with generating and verifying proofs.
+
+At first glance, recursive proofs may seem more complex and time-consuming. However, there are several benefits to this approach:
+
+. *Parallelization*: Recursive proofs allow for work parallelization, reducing user latency and improving SHARP efficiency.
+. *Lower cloud costs*: Since each job is shorter, the required memory for processing is reduced, resulting in lower cloud costs.
+. *Optimization*: Recursive proofs enable the SHARP to optimize for various factors, including latency, on-chain costs, and time to proof.
+
+[NOTE]
+====
+Latency in Starknet encompasses the time taken for processing, confirming, and including transactions in a block. It is affected by factors like network congestion, transaction fees, and system efficiency. Minimizing latency ensures faster transaction processing and user feedback.
+
+Time to proof, however, specifically pertains to the duration required to generate and verify cryptographic proofs for transactions or operations.
+====
+
+== SHARP Backend Architecture and Data Pipeline
+
+The SHARP back end architecture consists of several services that work together to process Cairo jobs and generate proofs. These services include:
+
+. *Gateway*: Cairo jobs enter the SHARP through the gateway.
+. *Validator*: This is the first important step. The validator service runs validation checks on each job, ensuring they meet the requirements and can fit within the prover machines. Invalid jobs are tagged as such and do not proceed to the Prover.
+. *Scheduler*: The scheduler service creates "trains" that aggregate jobs and send them to the Prover. Recursive jobs are paired and sent to the Prover together.
+. *Cairo Runner*: This service runs Cairo for the Prover’s needs. The Cairo Runner service runs Cairo programs, executing the necessary computations and generating the execution trace as an intermediate result. The Prover then uses this execution trace.
+. *Prover*: The Prover computes the proofs for the jobs.
+. *Dispatcher*: The dispatcher sends the proofs to either the Cairo Verifier for further recursion or the Solidity Verifier on Ethereum for final verification.
```

Comment:
Actually the dispatcher doesn't send the proof. It does 2 things:
In case of a proof that goes on chain - it creates what is called 'packages' from the proof, that will be eventually sent on chain by a different service which is not mentioned here ('blockchain writer')
In case of a recursive proof it runs the cairo verifier program on the proof it just got from the prover, which is resulting in a new cairo job that goes back to the validator.
(it used to do only the first part I mentioned, and when we added the recursion flow we added the second funcitonality to the dispacher service as well even though 'dispatcher' now is a bit confusing, because it doesn't 'dispatch' anything in this flow)